### PR TITLE
feature/MARP-278-Artifact-selector-must-display-unique-option

### PR DESCRIPTION
**Pull request by nntthuy-axonivy**

### CODE REVIEW

File: src/app/pages/product/ProductView.php [hunk 1]
```diff
@@ -103,16 +103,30 @@ public function getMavenArtifactsAsDependency(): array
       return [];
     }
     $mavenArtifacts = $this->mavenProductInfo->getMavenArtifacts($this->version);
-    return array_filter($mavenArtifacts, fn(MavenArtifact $artifact) => $artifact->getMakesSenseAsMavenDependency());
+    return array_filter($mavenArtifacts, fn (MavenArtifact $artifact) => $artifact->getMakesSenseAsMavenDependency());
```
Comment (tutn-axonivy) [inline src/app/pages/product/ProductView.php]: remove the space after fn

**Review by tutn-axonivy — APPROVED** [review]
LGTM

**Comment by nntthuy-axonivy** [comment]
Hi @alexsuter, currently artifact dropdown has duplicated names because name in meta.json and product.json are the same. To prevent this issue, I filter list of artifacts based on their name and type before rendering them in UI.
Please have a look on this PR. Thank you!

**Comment by alexsuter** [comment]
Please add a test.

**Review by alexsuter — COMMENTED** [review]
Please add a test

**Comment by alexsuter** [comment]
Can you show me in the axon ivy market which one has duplicated artifacts at the moment?

**Comment by nntthuy-axonivy** [comment]
> Can you show me in the axon ivy market which one has duplicated artifacts at the moment?

Hi @alexsuter, "Adobe Acrobat Connector" has 3 duplicated artifact names
image
I haven't finished writing tests, just committed new codes to check CI build. I'll inform you when PR is ready to review.

**Comment by alexsuter** [comment]
I think this was not the case before you have merged some other PRs. What's the reason why this happen now?

**Comment by nntthuy-axonivy** [comment]
> I think this was not the case before you have merged some other PRs. What's the reason why this happen now?

oh I do not notice that it was correct before. Let me check which PR affects this case

**Comment by nntthuy-axonivy** [comment]
Hi @alexsuter, I discovered that some PRs are affecting the code. My proposed solution for Adobe Acrobat Connector is to remove the "archiveArtifacts" block in meta.json, add a "-demo" suffix to the `artifactId` in `maven-import`, and then close the PR. What are your thoughts?

**Review by alexsuter — CHANGES_REQUESTED** [review]
I think this PR solves an issue which has been introduced before. Therefore I would fix the real problem. Rather than implementing new hacks.

**Comment by alexsuter** [comment]
Closing this PR and rename the branch to stale/* to have less builds on our build server.

**Pull request closed without merging**